godot-core/src/lib.rs

```diff
@@ -77,14 +79,41 @@ pub mod private {
         F: FnOnce() -> R + std::panic::UnwindSafe,
         S: std::fmt::Display,
     {
-        match std::panic::catch_unwind(code) {
+        let location_line: Arc<Mutex<u32>> = Arc::new(Mutex::new(0));
+        let location_file: Arc<Mutex<String>> = Arc::new(Mutex::new(String::from("file")));
+
+        // Back up previous hook, set new one
+        let prev_hook = std::panic::take_hook();
+        {
+            let location_line = location_line.clone();
+            let location_file = location_file.clone();
+            std::panic::set_hook(Box::new(move |panic_info| {
+                if let Some(location) = panic_info.location() {
+                    *location_line.lock().unwrap() = location.line();
+                    *location_file.lock().unwrap() = String::from(location.file());
```

Comment:
I dont think this will in practice ever fail to unwrap, but i'd probably suggest using `expect` here instead of `unwrap`. Especially since im not fully sure what happens if we panic inside of a panic hook, https://stackoverflow.com/questions/73467248/what-happens-when-a-panic-hook-panics this seems to say that the process aborts which isn't ideal but it's fine. 

So it'd be nice to document why this is likely gonna be ok to do.

Reply:
Thinking about it a bit more, i'd probably suggest not unwrapping here and instead failing by not writing anything to `location_file`/`location_line`. Since then we at least get a nicer error message instead of the process aborting.

Reply:
Using `lock().unwrap()` is a very common idiom, I'd definitely keep it that way.
It never fails in practice, it's just to protect against lock poisoning.

See also:
* [Should I unwrap() a mutex lock?](https://users.rust-lang.org/t/should-i-unwrap-a-mutex-lock/61519)
* [`parking_lot::Mutex::lock()`](https://docs.rs/lock_api/latest/lock_api/struct.Mutex.html#method.lock), which directly returns the guard